ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs

```diff
@@ -261,12 +266,33 @@ public int OnAfterCloseSolution(object aPUnkReserved)
       return VSConstants.S_OK;
     }
 
-
     #endregion
 
 
     #region Private Methods
 
+    private bool SolutionDoesNotContainCppProject()
+    {
+      DTE2 dte2 = (DTE2) Package.GetGlobalService(typeof(DTE));
+      var solution = dte2.Solution;
+
+      if (solution == null)
+      {
+        return true;
+      }
+
+      foreach (var project in solution)
+      {
+        var proj = (Project) project;
+        if (proj.FullName.EndsWith(ScriptConstants.kVcxprojExtension))
```

Comment:
For project type detection is recommended to check the project GUID.

Reply:
You're right, I'll try to find a way to check the project GUID.